src/components/popover/popover.ts

```diff
@@ -1,112 +1,303 @@
    */
-  flipHorizontalOffset: Pixel;
+  anchorRect: Rect;
+
   /**
-   * The component is moved by this amount to the top when
-   * it is rendered on the top.
+   * Rectangle inside which the popover should stay when being displayed. It's the Rect of the "o-spreadsheet"
+   * element by default.
+   *
+   * Coordinates are expressed relative to the "o-spreadsheet" element.
    */
-  flipVerticalOffset: Pixel;
-  onMouseWheel: (ev: WheelEvent) => void;
+  containerRect?: Rect;
+
+  /** The popover can be positioned below the anchor Rectangle, or to the right of the rectangle */
+  positioning: PopoverPosition;
+
+  maxWidth?: Pixel;
+  maxHeight?: Pixel;
+
+  /** Offset to apply to the vertical position of the popover.*/
+  verticalOffset: number;
+
+  onMouseWheel?: () => {};
 }
 
-export class Popover extends Component<Props, SpreadsheetChildEnv> {
+css/* scss */ `
+  .o-popover {
+    position: absolute;
+    z-index: ${ComponentsImportance.Popover};
+    overflow: auto;
+    box-shadow: 1px 2px 5px 2px rgb(51 51 51 / 15%);
+    width: fit-content;
+    height: fit-content;
+  }
+`;
+
+export class Popover extends Component<PopoverProps, SpreadsheetChildEnv> {
   static template = "o-spreadsheet-Popover";
   static defaultProps = {
-    flipHorizontalOffset: 0,
-    flipVerticalOffset: 0,
+    positioning: "BottomLeft",
     verticalOffset: 0,
-    marginTop: 0,
     onMouseWheel: () => {},
   };
 
-  private spreadsheetPosition = useSpreadsheetPosition();
-
-  get style() {
-    // the props's position is expressed relative to the "body" element
-    // but we teleport the element in ".o-spreadsheet" to keep everything
-    // within our control and to avoid leaking into external DOM
-    const horizontalPosition = `left:${this.horizontalPosition() - this.spreadsheetPosition.x}`;
-    const verticalPosition = `top:${this.verticalPosition() - this.spreadsheetPosition.y}`;
-    const height = `max-height:${
-      this.viewportDimension.height - BOTTOMBAR_HEIGHT - SCROLLBAR_WIDTH
-    }`;
-    return `
-      position: absolute;
-      z-index: ${ComponentsImportance.Popover};
-      ${verticalPosition}px;
-      ${horizontalPosition}px;
-      ${height}px;
-      width:${this.props.childWidth}px;
-      overflow-y: auto;
-      overflow-x: hidden;
-      box-shadow: 1px 2px 5px 2px rgb(51 51 51 / 15%);
-    `;
-  }
-
-  private get viewportDimension(): DOMDimension {
-    return this.env.model.getters.getSheetViewDimensionWithHeaders();
-  }
-
-  private get shouldRenderRight(): boolean {
-    const { x } = this.props.position;
-    return x + this.props.childWidth < this.viewportDimension.width;
-  }
-
-  private get shouldRenderBottom(): boolean {
-    const { y } = this.props.position;
+  private popoverRef = useRef("popover");
+
+  setup() {
+    // useEffect occurs after the DOM is created and the element width/height are computed, but before
+    // the element in rendered, so we can still set its position
+    useEffect(() => {
```

Comment:
Do you think having `el/el.parent.getBoundingClientRect()` as dependency is fine? I feel like it might mess up

Reply:
And (Owl noo here) is there a better way to create dependencies than that ? 

```suggestion
      () => [
        ...Object.values(this.props.anchorRect),
        ...Object.values(this.props.containerRect || {}),
        this.props.maxHeight,
        this.props.maxWidth,
        this.props.positioning,
        this.props.verticalOffset,
        ...Object.values(this.popoverRef.el?.getBoundingClientRect() || {}),
        ...Object.values(this.popoverRef.el?.parentElement?.getBoundingClientRect() || {}),
      ]
```
? I cannot put the object themselves in the dependancies since owl check if the references are the same, not the values